scheduler/src/cook/kubernetes/api.clj

```diff
@@ -655,13 +655,29 @@
           (.setVolumeMounts container [(workdir-volume-mount-fn true) (sidecar-workdir-volume-mount-fn false)])
           (.addContainersItem pod-spec container))))
 
-    (when hostname
+    (if hostname
       ; Why not just set the node name?
       ; The reason is that setting the node name prevents pod preemption
       ; (https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/)
       ; from happening. We want this pod to preempt lower priority pods
       ; (e.g. synthetic pods).
-      (add-node-selector pod-spec k8s-hostname-label hostname))
+      (add-node-selector pod-spec k8s-hostname-label hostname)
+      (when (seq pod-hostnames-to-avoid)
+        ; Use node "anti"-affinity to disallow scheduling
+        ; on hostnames we're being told to avoid
+        (let [affinity (V1Affinity.)
```

Comment:
Nitpick: Is there a link to some documentation explaining what we're setting up below?
It'd be nice to include in a comment here.

Reply:
Good call, added a link to [this](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#node-affinity).